torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3167,6 +3177,60 @@ def sample_inputs_take_along_dim(op_info, device, dtype, requires_grad, **kwargs
             )
 
 
+def error_inputs_aminmax_amax_amin(op_info, device, **kwargs):
+
+    # Error Inputs for zero-dim tensors, when 'dim' arg is not provided.
+    shape = (S, 0, S)
+    err_msg_amax_amin = "Specify the reduction dim with the 'dim' argument."
+    err_msg_aminmax = "cannot compute aminmax over an empty dimension as the operation has no identity"
+    if op_info in ['amax', 'amin']:
+        yield ErrorInput(SampleInput(torch.rand(shape, device=device)), error_type=RuntimeError,
+                         error_regex=err_msg_amax_amin)
+    elif op_info in ['aminmax']:
+        yield ErrorInput(SampleInput(torch.rand(shape, device=device)), error_type=RuntimeError,
+                         error_regex=err_msg_aminmax)
+
+    # Error Inputs for tensors with more than 64 dimension
+    sizes = [1] * 65
+    err_msg1 = "only tensors with up to 64 dims are supported"
+    yield ErrorInput(SampleInput(torch.randn(sizes, device=device), kwargs={'dim': -1}),
+                     error_type=RuntimeError, error_regex=err_msg1)
+    yield ErrorInput(SampleInput(torch.randn(sizes, device=device), kwargs={'dim': 64}),
+                     error_type=RuntimeError, error_regex=err_msg1)
+
+    # Error Inputs for repeated 'dim'
+    if op_info in ['amax', 'amin']:
+        dims = [(0, 0), (0, -4)]
+        err_msg2 = "dim 0 appears multiple times in the list of dims"
+        x = torch.randn(S, S, S, S, device=device)
+        for dim in dims:
+            yield ErrorInput(SampleInput(x, kwargs={'dim': dim}), error_type=RuntimeError,
+                             error_regex=err_msg2)
+
+    # Error Input for illegal dtype
+    input5 = torch.randn(L, L, dtype=torch.float32, device=device)
+    max_values = torch.empty(L, dtype=torch.float32, device=device)
+    min_values = torch.empty(L, dtype=torch.double, device=device)
+    illegal_values = torch.empty(L, dtype=torch.int, device=device)
+
+    err_msg_amax_amin2 = "Expected the dtype for input and out to match"
+    err_msg_aminmax2 = "Expected out tensor to have dtype float, but got int instead"
+
+    if op_info in ['amax', 'amin']:
+        yield ErrorInput(SampleInput(input5, kwargs={'dim': 0, 'out': illegal_values}),
+                         error_type=RuntimeError,
+                         error_regex=err_msg_amax_amin2)
+    elif op_info in ['aminmax']:
+        yield ErrorInput(SampleInput(input5, kwargs={'dim': 0, 'out': (max_values, min_values)}),
+                         error_type=RuntimeError, error_regex=err_msg_aminmax2)
+
+    # Error Inputs for functions to raise an error on specified zero'd dimension as reduction dim
+    err_msg_amax_amin3 = "Dimension out of range"
+    err_msg_aminmax3 = "Expected reduction dim 1 to have non-zero size"
+    yield ErrorInput(SampleInput(torch.rand(shape, device=device), kwargs={'dim': 1}),
```

Comment:
why do you expect `dimension out of range` error here? I see "Expected reduction dim 1 to have non-zero size" for amin/amax also

Reply:
My fault. Corrected that